src/charm.py

```diff
@@ -94,6 +115,35 @@ def _pebble_layer(self) -> Dict:
             },
         }
 
+    def _reset_instance_action(self, event: ActionEvent) -> None:
+        """Reset instance and report action result.
+
+        Args:
+            event: Event triggering the reset instance action.
+        """
+        results = {
+            "reset-instance": False,
+        }
+        if not self.model.unit.is_leader():
+            event.defer()
```

Comment:
@gregory-schiano I'm not sure on this, should this just do nothing unless the unit is the leader? I'm not sure if all the units get the action or just a specific unit?

Reply:
I guess:
* If all units get the action -> do nothing if the unit isn't the leader
* If only a specific unit gets the action -> fail the action if it isn't the leader

Not sure what deferring would do to an operator generated action

Reply:
In the case of Synapse, for now, this check is just in case...Scaling will not be allowed and will be addressed in a different story/PR.

Reply:
The event will fail with the message "Only the juju leader unit can run reset instance action" now.